### PR TITLE
Simplify parameterized host functionality

### PR DESCRIPTION
**Pull request by catalinaperalta**
This PR simplifies the parameterized host functionality and consolidates client param grouping in the transform.

### CODE REVIEW

File: test/autorest/generated/morecustombaseurigroup/client.go [hunk 1]
```diff
@@ -53,13 +53,13 @@ func (c *ClientOptions) telemetryOptions() azcore.TelemetryOptions {
...
 type Client struct {
-	dnsSuffix      string
 	subscriptionID string
+	dnsSuffix      string
 	p              azcore.Pipeline
 }
 
 // NewClient creates an instance of the Client type with the specified endpoint.
-func NewClient(dnsSuffix *string, subscriptionID string, options *ClientOptions) (*Client, error) {
+func NewClient(subscriptionID string, dnsSuffix *string, options *ClientOptions) (*Client, error) {
```
Comment (jhendrixMSFT) [inline test/autorest/generated/morecustombaseurigroup/client.go:62]: SubscriptionID is in the wrong place, it should be an operation group method param.

Comment (catalinaperalta) [reply test/autorest/generated/morecustombaseurigroup/client.go]: Fixed

File: test/autorest/generated/custombaseurlgroup/paths.go [hunk 2]
```diff
@@ -44,8 +44,8 @@ func (client *pathsOperations) GetEmpty(ctx context.Context, accountName string)
 // getEmptyCreateRequest creates the GetEmpty request.
 func (client *pathsOperations) getEmptyCreateRequest(accountName string) (*azcore.Request, error) {
 	host := "http://{accountName}{host}"
+	host = strings.ReplaceAll(host, "{host}", client.host)
```
Comment (catalinaperalta) [inline test/autorest/generated/custombaseurlgroup/paths.go:47]: I think this is being generated incorrectly...shouldn't it be `client.Client.host`?
Also surprised that the tests pass with this...or am I missing something?

File: test/autorest/generated/morecustombaseurigroup/paths.go [hunk 3]
```diff
@@ -45,12 +45,12 @@ func (client *pathsOperations) GetEmpty(ctx context.Context, vault string, secre
 // getEmptyCreateRequest creates the GetEmpty request.
 func (client *pathsOperations) getEmptyCreateRequest(vault string, secret string, keyName string, pathsGetEmptyOptions *PathsGetEmptyOptions) (*azcore.Request, error) {
 	host := "{vault}{secret}{dnsSuffix}"
+	host = strings.ReplaceAll(host, "{dnsSuffix}", client.dnsSuffix)
```
Comment (catalinaperalta) [inline test/autorest/generated/morecustombaseurigroup/paths.go:48]: Same as above...shouldn't this be `client.Client.dnsSuffix`?

File: test/autorest/generated/custombaseurlgroup/paths.go (hunk 2 above)
Comment (jhendrixMSFT) [reply test/autorest/generated/custombaseurlgroup/paths.go:47]: The `*Client` is anonymously embedded in `pathsOperations`, so we can access it directly.

Comment (catalinaperalta) [reply test/autorest/generated/custombaseurlgroup/paths.go:47]: Ah true, a mistake I made in the previous version. Thanks for fixing it!

**Pull request merged**